packages/data-api/src/validators.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { hasNoContent, takesDateForm, takesIdForm, yup, yupUtils } from '@tupaia/utils';
+
+export const eventOptionsValidator = yup.object().shape({
+  organisationUnitCodes: yup.array().of(yup.string().required()).strict().required(),
+  startDate: yup
+    .string()
+    .test(
+      yupUtils.yupTestAny([hasNoContent, takesDateForm], 'startDate should be in ISO 8601 format'),
```

Comment:
Introduced this `yupUtils.yupTestAny([...tests], message?)` function as `yup` has not support for a validator that passes if one of its tests passes. In theory we might want a `yupTestAll` function at some point, but happy to leave that for now.